tuning.py

```diff
@@ -0,0 +1,317 @@
+                       epistatic_horizon: float,
+                       field_scale: float = 1.0,
+                       single_mut_offset: float = 0.0,
+                       epi_offset: float = 0.0) -> float:
+    """Returns the coupling scale tuning parameter to achieve `epistatic_horizon`.
+
+    The epistatic horizon depends on other landscape statistics.
+    Let $s_+$ be the average adaptive single mutant effect.
+    Let $e_{+,+}$ be the average epistatic effect for a pair of adaptive singles.
+    Then, given K, we solve the following equation to determine the coupling-scale:
+
+    $$K * field_scale (s_+ + field-scale) + (K choose 2) * coupling-scale (e_{+,+} + epi_offset) = 0$$
+    """
+    adaptive_threshold = 0.0
+    is_adaptive = True
+
+    pretuned_landscape = potts_model.PottsModel(landscape.weight_matrix,
+                                                landscape.field_vec,
+                                                wt_seq=landscape.wildtype_sequence,
+                                                field_scale=field_scale,
+                                                single_mut_offset=single_mut_offset,
+                                                epi_offset=epi_offset)
+    mean_adaptive_single_effect = get_mean_single_effect(
+        pretuned_landscape, threshold=adaptive_threshold, adaptive=is_adaptive)
+    mean_epistasis_effect = get_doubles_df(
+        pretuned_landscape, threshold=adaptive_threshold, adaptive=is_adaptive).residual.mean()
+
+    K = epistatic_horizon
+    numerator = - K * mean_adaptive_single_effect
+    denominator = comb(K, 2) * mean_epistasis_effect
+    coupling_scale = numerator / denominator
+    return coupling_scale
+
+
+# TODO(nthomas) implement fraction_reciprocal_deleterious_epistasis:
+#   The fraction of deleterious(-, -) doubles that exhibit positive epistasis.
+def get_tuning_kwargs(landscape: potts_model.PottsModel,
+                      fraction_adaptive_singles: Optional[float] = None,
+                      fraction_reciprocal_adaptive_epistasis: Optional[float] = None,
+                      epistatic_horizon: Optional[float] = None,
+                      normalize_to_singles: bool = False) -> Dict[str, float]:
+    """Returns the landscape tuning parameters.
+
+    Args:
+      landscape: A landscape.
+      fraction_adaptive_singles: The fraction of singles that achieve a fitness > wildtype. If unset,
+        this fraction is preserved in the initial landscape
+      fraction_reciprocal_adaptive_epistasis: The fraction of adaptive (+, +) doubles that exhibit negative
+        epistasis.
+      epistatic_horizon: average distance at which epi effects out weigh singles
+      normalize_to_singles: A boolean, when True, ensures that the standard deviation of single mutant effects
+        is 1.0.
+
+    Returns:
+      A dict of tuning parameters
+      A tuple of [shift, shift, scale, scale] tuning parameters.
+    """
+    # default tuning params
```

Comment:
I found 'default' confusing here. Does the output of the function depend on the choice of these?

Reply:
Yeah this is related to the above comment about leaving some properties untuned. Since the epistatic horizon depends on all of the other tuning parameters, these will affect the final output. Maybe it would be easier to read if I add them to `if` `else` clauses instead of adding defaults here.